### PR TITLE
Merge InitShardPrimary functionality into PlannedReparentShard

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR merges the functionality of InitShardPrimary and PlannedReparentShard, so that just calling PRS even for initialisation would be sufficient.

However, PlannedReparentShard would not do the destructive step of copying the executed gtid set from the new primary to all the other replicas. PlannedReparentShard is meant to be used for non-destructive repairable operations. For this use case, it is still advised that users use InitShardPrimary.

## Related Issue(s)

- #8975 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required Documentation is required to say that PRS should be goto option to initialise the cluster

### CODE REVIEW

File: go/test/endtoend/vreplication/cluster.go
Comment (deepthi) [inline go/test/endtoend/vreplication/cluster.go:350]: Should we log InitializeShard instead of PRS?

File: go/vt/vtctl/reparentutil/planned_reparenter.go
Comment (deepthi) [inline go/vt/vtctl/reparentutil/planned_reparenter.go:520]:
```suggestion
	// needsReload is used to keep track of whether we need to refresh the state
```
```suggestion
	// needsReload is used to keep track whether we need to refresh the state
```

File: go/vt/vtctl/reparentutil/planned_reparenter.go
Comment (deepthi) [inline go/vt/vtctl/reparentutil/planned_reparenter.go:528]: Let's rename this to `needsRefresh`. reload makes me think we are reloading topo information.

**Review by deepthi — APPROVED** [review]
LGTM. Nice work!
Let's wait for @sougou to give it a once-over before we merge.
Can you fix the little things I found in the meantime?

File: go/vt/vtctl/reparentutil/planned_reparenter.go
Comment (GuptaManan100) [reply go/vt/vtctl/reparentutil/planned_reparenter.go]: fixed

File: go/vt/vtctl/reparentutil/planned_reparenter.go
Comment (GuptaManan100) [reply go/vt/vtctl/reparentutil/planned_reparenter.go]: fixed

File: go/test/endtoend/vreplication/cluster.go
Comment (GuptaManan100) [reply go/test/endtoend/vreplication/cluster.go]: fixed

**Review by sougou — APPROVED** [review]

**Pull request merged**
